Convert package to ROS2

timrobotson opened the pull request:
- Remove all references to Turtlebot as it was only included for backwards-compatibility reasons. This package should be robot agnostic.
- Convert launch files, CMakeLists and package.xml to ROS2 equivalent.

Tested running in a ROS2 dashing container running both launch files with and without `gui:=true` and it launched correctly in all cases. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

timrobotson committed:
Convert package to ROS2

- Remove all references to turtlebot as it was only included for
backwards-compatibility reasons. This package should be robot agnostic.
- Convert launch files, CMakeLists and package.xml to ROS2 equivilant.

ryanewel commented:
> * Remove all references to Turtlebot as it was only included for backwards-compatibility reasons. This package should be robot agnostic.

Does removing the turtlebot launch files have any impact on downstream dependencies? E.g. the navigation launch files have been deleted.

What are we doing with the turtelbot3 map files in this package? Did you get a chance to test those with ROS2?

timrobotson commented:
> Does removing the turtlebot launch files have any impact on downstream dependencies? E.g. the navigation launch files have been deleted.

We originally left the turtlebot files in because voice interaction was relying on them, but because this is a major bump we can remove them all and any ROS2 sample apps (or other apps using htis package) can import the robot themselves. 

> What are we doing with the turtelbot3 map files in this package? Did you get a chance to test those with ROS2?

I think we want the map files as they describe a path that the robot takes around the store right? I'm unsure if they are robot agnostic but it would be good to have a defined path that others can make their robot follow. @samuelgundry what do you think? 

I haven't tested the map files with ROS2. I'll test with cloudwatch sample application as I'm in the process of porting it's simulation_ws to ROS2 using this branch of bookstore-world.

timrobotson committed:
HACK: Make bookstore-world work with cloudwatch

timrobotson commented:
I've got this working with the ROS2 port of cloudwatch-sample-application however it has 2 hacks that need to be fixed before this is merged. It needs to set additional environment variables GAZEBO_MODEL_PATH and GAZEBO_RESOURCE_PATH so that it can correctly load all the models / mesh's for this simulation and the turtlebot. 

The hacks that need to be fixed are:
- The path to the default gazebo resources and models directories are hard coded, this should be dynamic somehow. The are loaded when sourcing `/usr/share/gazebo-9/setup.sh` on Ubuntu.
- The path to turtlebot3_description_reduced_mesh models has been added but this should be a dynamic parameter passed to this launch file instead of hard coded in there as this package should have no knowledge about turtlebot.

timrobotson committed:
Add ability to specify additional gazebo_model_paths

- When launching gzserver add ability to add additional gazebo model
paths so we can add it for the turtlebot etc without hard coding it into
the bookstore world.
